samza-sql/src/main/java/org/apache/samza/sql/impl/ConfigBasedUdfResolverFactory.java

```diff
@@ -0,0 +1,97 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.impl;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.samza.SamzaException;
+import org.apache.samza.sql.interfaces.udfs.ScalarUdf;
+import org.apache.samza.sql.interfaces.UdfMetadata;
+import org.apache.samza.sql.interfaces.UdfResolver;
+import org.apache.samza.sql.interfaces.UdfResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ConfigBasedUdfResolverFactory implements UdfResolverFactory {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ConfigBasedUdfResolverFactory.class);
+  public static final String CFG_UDF_CLASSES = "udfClasses";
+  public static final String UDF_METHOD_NAME = "execute";
+
+  @Override
+  public UdfResolver create(Properties config) {
+    return new ConfigBasedUdfResolver(config);
+  }
+
+  private class ConfigBasedUdfResolver implements UdfResolver {
+    private final ArrayList<UdfMetadata> udfs;
+
+    public ConfigBasedUdfResolver(Properties config) {
+      List<String> udfClasses = Arrays.stream(config.getProperty(CFG_UDF_CLASSES, "").split(","))
+          .filter(StringUtils::isNotBlank)
+          .collect(Collectors.toList());
+      udfs = new ArrayList<>();
+      Class<?> udfClass;
+      for (String udfClassName : udfClasses) {
+        try {
+          udfClass = Class.forName(udfClassName);
+        } catch (ClassNotFoundException e) {
+          String msg = String.format("Couldn't load the udf class %s", udfClassName);
+          LOG.error(msg, e);
+          throw new SamzaException(msg, e);
+        }
+
+        if (!ScalarUdf.class.isAssignableFrom(udfClass)) {
+          String msg = String.format("Udf class %s is not extended from %s", udfClassName, ScalarUdf.class.getName());
+          LOG.error(msg);
+          throw new SamzaException(msg);
+        }
+
+        Optional<Method> udfMethod =
```

Comment:
Do you allow any number of arguments, return type? If not, it would be best to validate them as early as possible - this looks like a good place.